File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    // changed arg numbers => n
    vector<int> twoSum(vector<int>& n, int target) {
        int start = 0, end = n.size() - 1;
        while (start <= end) {
            if (n[start] + n[end] == target) {
                return {start + 1, end + 1};
            } else if (n[start] + n[end] < target) {
                start++;
            } else if (n[start] + n[end] > target) {
                end--;
            } else {
                cout << "help!" << endl;
                break;
            }
        }
        // sum not found
        return {-1, -1};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    var i = 0;
    var j = numbers.length - 1;
    while (i < j) {
        var sum = numbers[i] + numbers[j];
        if (sum === target) {
            return [i + 1, j + 1];
        } else if (sum < target) {
            i++;
        } else {
            j--;
        }
    }
    return [null, null];
};